src/Parsers/ParserCreateUserQuery.h

```diff
@@ -7,13 +7,13 @@ namespace DB
 {
 /** Parses queries like
   * CREATE USER [IF NOT EXISTS | OR REPLACE] name
-  *     [IDENTIFIED [WITH {NO_PASSWORD|PLAINTEXT_PASSWORD|SHA256_PASSWORD|SHA256_HASH|DOUBLE_SHA1_PASSWORD|DOUBLE_SHA1_HASH}] BY {'password'|'hash'}]
+  *     [IDENTIFIED [WITH {NO_PASSWORD|PLAINTEXT_PASSWORD|SHA256_PASSWORD|SHA256_HASH|DOUBLE_SHA1_PASSWORD|DOUBLE_SHA1_HASH|LDAP}] BY {'password'|'hash'|'server_name'}]
```

Comment:
As far as I understood you're going to use another form

```
IDENTIFIED WITH ldap SERVER 'server_name' AS yyy
```

right?

Reply:
Initially, I wanted to use SERVER, but then decided to change as little as possible.
Maybe then we can use different syntax here, like: `IDENTIFIED USING ldap SERVER 'server_name'`?

Reply:
Using `LDAP_SERVER` everywhere, so the syntax will look like this now:
```sql
... IDENTIFIED WITH LDAP_SERVER BY 'server_name' ...
```
Let me know, what do you think about this.

Reply:
Lowercase would be more consistent here:

```
IDENTIFIED WITH ldap_server BY 'server_name'
```

Reply:
Currently, it is in lower case. This conversation points to an outdated version of the code.